packages/contracts/contracts/Unirep.sol

```diff
@@ -388,6 +388,43 @@ contract Unirep is IUnirep, VerifySignature {
         attester.currentEpoch = newEpoch;
     }
 
+    function verifyEpochKeyProof(
+        uint256[] memory publicSignals,
+        uint256[8] memory proof
+    ) public returns (bool) {
+        bool valid = epochKeyVerifier.verifyProof(publicSignals, proof);
+        // short circuit if the proof is invalid
+        if (!valid) return false;
+        require(publicSignals[3] < type(uint160).max);
+        require(publicSignals[0] < maxEpochKey);
+        updateEpochIfNeeded(uint160(publicSignals[3]));
+        AttesterData storage attester = attesters[uint160(publicSignals[3])];
+        // epoch check
+        if (publicSignals[2] > attester.currentEpoch) return false;
+        // state tree root check
+        if (!attester.stateTreeRoots[publicSignals[2]][publicSignals[1]])
+            return false;
+        return true;
+    }
+
+    function verifyReputationProof(
+        uint256[] memory publicSignals,
+        uint256[8] memory proof
+    ) public returns (bool) {
+        bool valid = reputationVerifier.verifyProof(publicSignals, proof);
+        if (!valid) return false;
+        require(publicSignals[3] < type(uint160).max);
+        require(publicSignals[0] < maxEpochKey);
+        updateEpochIfNeeded(uint160(publicSignals[3]));
+        AttesterData storage attester = attesters[uint160(publicSignals[3])];
+        // epoch check
+        if (publicSignals[2] > attester.currentEpoch) return false;
+        // state tree root check
+        if (!attester.stateTreeRoots[publicSignals[2]][publicSignals[1]])
+            return false;
+        return true;
+    }
+
```

Comment:
I think `verifyUSTProof`, `verifySignUpProof`, `verifyAggregateEpochKeysProof` can also be functions in `Unirep.sol`
the public signals can also be verified in these functions

Reply:
Yea, I think we can do this as a part of #157